xbmc/dialogs/GUIDialogKaiToast.cpp

```diff
@@ -134,6 +135,9 @@ bool CGUIDialogKaiToast::DoWork()
     //  Play the window specific init sound for each notification queued
     SetSound(toast.withSound);
 
+    // Activate haptics for this notification
+    PERIPHERALS::g_peripherals.OnUserNotification();
```

Comment:
Maybe this should only be done when the notification doesn't follow right after a previous notification. Otherwise when e.g. updating Kodi to a new version and it start updating all add-ons the controller won't stop rumbling for quite a while which certainly also affects its battery.